### PR TITLE
MMapRegion: Split between typed and untyped, handle error

### PR DESCRIPTION
**Pull request by Geod24**
Slight refactoring and documentation of MMapRegion in order to make it less awkward to use for consumer that have no need for a typed mapping.

### CODE REVIEW

File: src/core/internal/elf/io.d [hunk 1]
```diff
@@ -283,37 +284,65 @@ version (LinuxOrBSD)
...
-
-        auto ptr = cast(ubyte*) mmap(null, mappedSize, PROT_READ, MAP_PRIVATE, fd, pagedOffset);
+        const offset = pageOffset * pageSize;
+        const mappedSize = pageCount * pageSize;
 
-        mappedRegion = ptr[0 .. mappedSize];
-        data = cast(T*) (ptr + offsetDiff);
+        auto ptr = mmap(null, mappedSize, PROT_READ, MAP_PRIVATE, fd, offset);
+        if (ptr !is MAP_FAILED)
+            this.data = (cast(const(ubyte)*) ptr)[0 .. mappedSize];
```
Comment (PetarKirov) [inline src/core/internal/elf/io.d:333]: I think you missed one of my suggestions:

```D
        const ptr = cast(ubyte*) mmap(null, mappedSize, PROT_READ, MAP_PRIVATE, fd, offset);
        if (ptr !is MAP_FAILED)
            this.data = ptr[0 .. mappedSize];
 ```

Comment (Geod24) [reply src/core/internal/elf/io.d]: I just wasn't so found of it, but suggestion applied nonetheless :)

**Comment by Geod24** [comment]
@PetarKirov : All comments should have been addressed (selective import and the different `cast`).

**Comment by PetarKirov** [comment]
Thanks @Geod24, I've set the `auto-merge` label.
P.S. And sorry for the delay - please message me directly on Slack, as I have too many GitHub notifications and I don't always manage to catch up with all of them in a timely manner.

**Pull request merged**